specs/AppLifecycle/Restart/restartApi.markdown

```diff
@@ -0,0 +1,259 @@
+</li>
+<li>
+<p>2: RESTART_NO_HANG - Do not restart the process if it terminates due to the application not responding.</p>
+</li>
+<li>
+<p>4: RESTART_NO_PATCH - Do not restart the process if it terminates due to the installation of an OS update.</p>
+</li>
+<li>
+<p>8: RESTART_NO_REBOOT - Do not restart the process if the computer is restarted as the result of an OS update.</p>
+</li>
+</ul></td>
+</tr>
+<tr class="even">
+<td>HRESULT <strong>UnregisterApplicationRestart</strong>();</td>
+<td>Unregisters the app for restart, where the app had previously called RegisterApplicationRestart.</td>
+</tr>
+<tr class="odd">
+<td>public static IAsyncOperation&lt;AppRestartFailureReason&gt; <strong>RequestRestartAsync</strong>(<br />
+string launchArguments);</td>
+<td><p>Static function in CoreApplication. Requests immediate termination and restart.</p>
+<ul>
+<li>
+<p>The launchArguments string is passed into the new instance as the Arguments property on LaunchActivatedEventArgs.</p>
+</li>
+<li>
+<p>Also sets LaunchActivatedEventArgs.PreviousExecutionState to Terminated.</p>
+</li>
+</ul></td>
+</tr>
+<tr class="even">
+<td>public enum <strong>AppRestartFailureReason</strong></td>
+<td><ul>
+<li>
+<p>0: RestartPending - a restart is already in progress.</p>
+</li>
+<li>
+<p>1: NotInForeground - an app must be visible and in the foreground when it calls the restart API.</p>
+</li>
+<li>
+<p>2: InvalidUser - could not restart for the specified user.</p>
+</li>
+<li>
+<p>3: Other - unspecified failure.</p>
+</li>
+</ul></td>
+</tr>
+<tr class="odd">
+<td>class <strong>LaunchActivatedEventArg</strong></td>
+<td><p>Relevant Properties:</p>
+<ul>
+<li><p>Arguments: gets the arguments that are passed to the app during its launch activation.</p></li>
+<li><p>Kind: Gets the reason that this app is being activated. This is of type enum ActivationKind</p></li>
+</ul>
+<p>PreviousExecutionState: Gets the execution state of the app before this activation.</p></td>
+</tr>
+</tbody>
+</table>
+
+## New API Details
+
+### RequestRestartNow
+
+<table>
+<colgroup>
+<col style="width: 41%" />
+<col style="width: 53%" />
+<col style="width: 5%" />
+</colgroup>
+<thead>
+<tr class="header">
+<th>New API </th>
+<th>Description</th>
+</tr>
+</thead>
+<tbody>
+<tr class="odd">
+<td>static AppRestartFailureReason <strong>RequestRestartNow</strong>(string restartArguments)</td>
+<td><p>Equivalent to CoreApplication.RequestRestartAsync, except that it is <em>not</em> async.
+<ul>
+<li>If the restart fails, but the user subsequently launches the app manually, the app will launch normally and no restart arguments will be passed. </li>
+<li>If the app has any in-process background tasks running when it calls this API, those tasks will be cancelled in the normal way. Out-of-process background tasks will not be affected. </li>
+</ul>
+ The restarted instance will be activated with Windows.ApplicationModel.Activation.LaunchActivatedEventArgs.</p>
```

Comment:
That's a CoreApplication API, what happens in a Win32 app?

Reply:
DISCUSSION: The app is just CreateProcess'd with the string, and your WinMain/wmain get the string to process. So you'd say "/restart" and the API will invoke "contoso.exe /restart"

Reply:
So update to explain that rather than referring to LaunchActivatedEventArgs?

Reply:
The command-line is exposed via LaunchActivatedEventArgs, the object is returned via the GetActivatedEventArgs() method on the same object this API is exposed from.